src/main/java/com/example/onlinebookstore/service/BookServiceImpl.java

```diff
@@ -33,7 +34,25 @@ public List<BookDto> findAll() {
     @Override
     public BookDto findById(Long id) {
         Book book = bookRepository.findById(id)
-                .orElseThrow(() -> new EntityNotFoundException("Can't find book with id:" + id));
+                .orElseThrow(() -> new EntityNotFoundException(EXCEPTION_MSG_CANNOT_FIND + id));
         return bookMapper.toDto(book);
     }
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto requestDto) {
+        Book book = bookRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(EXCEPTION_MSG_CANNOT_FIND + id));

```

Comment:
Looks like dublicate code from .findById() method
And in the future, I think, it will be necessary to ensure the atomicity of this method

Reply:
Good point. You can consider creating a separate method which returns Book entity and checks for existence. But it's not critical now.